CrossSectionAnalysis/CrossSectionAnalysis.py

```diff
@@ -0,0 +1,422 @@
+        return
+    point = self.pathArray[int(value)]
+    direction = self.pathArray[int(value) + 1] - point
+    self.reformatLogic.SetSliceOrigin(self.inputSliceNode, point[0], point[1], point[2])
+    self.reformatLogic.SetSliceNormal(self.inputSliceNode, direction[0], direction[1], direction[2])
+    self.lastValue = value
+
+  def selectNode(self, inputPath):
+    # Observe the selected markup path only. Remove from previous.
+    self.removeMarkupObservers()
+    self.inputPath = inputPath
+    self.resetSliceNodeOrientationToDefault()
+    self.fillPathArray()
+    self.addMarkupObservers()
+    
+  def selectView(self, sliceMRMLNodeName):
+    self.inputSliceNode = slicer.util.getNode(sliceMRMLNodeName)
+    slicer.modules.reformat.widgetRepresentation().setEditedNode(slicer.util.getNode(sliceMRMLNodeName))
+    
+  def addMarkupObservers(self):
+    # Observe markup curve. VMTK centerlines don't seem to have UI handles.
+    if self.inputPath is not None and self.inputPath.GetClassName() == "vtkMRMLMarkupsCurveNode":
+        self.markupPointObserver = self.inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointEndInteractionEvent, self.onMarkupPointEndInteraction)
+        self.markupPointRemovedObserver = self.inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointRemovedEvent, self.onMarkupPointRemoved)
+        self.markupPointAddedObserver = self.inputPath.AddObserver(slicer.vtkMRMLMarkupsNode.PointAddedEvent, self.onMarkupPointAdded)
+        
+  def removeMarkupObservers(self):
+    if self.inputPath is not None:
+        self.inputPath.RemoveObserver(self.markupPointObserver)
+        self.inputPath.RemoveObserver(self.markupPointRemovedObserver)
+        self.inputPath.RemoveObserver(self.markupPointAddedObserver)
+        
+  # Reposition the slice if a markup control point is moved
+  def onMarkupPointEndInteraction(self, caller, event):
+    self.fillPathArray()
+    self.process(self.lastValue)
+    
+  # Reposition the slice to start if a markup control point is removed
+  def onMarkupPointRemoved(self, caller, event):
+    self.fillPathArray()
+    self.process(0)
+
+  # Reposition the slice to start if a markup control point is added
+  def onMarkupPointAdded(self, caller, event):
+    self.fillPathArray()
+    self.process(0)
+    
+  # Get RAS current position on path
+  def currentPosition(self, value):
+    if self.pathArray.size == 0:
+        return numpy.zeros(0)
+    return self.pathArray[int(value)]
+  
+  # Calculate distance of each point from start of path
+  def cumulateDistances(self):
+    self.cumDistancesArray = numpy.zeros(int(self.pathArray.size / 3))
+    previous = self.pathArray[0]
+    dist = 0
+    for i, point in enumerate(self.pathArray):
+      # https://stackoverflow.com/questions/1401712/how-can-the-euclidean-distance-be-calculated-with-numpy
+      dist += numpy.linalg.norm(point - previous)
+      self.cumDistancesArray[i] = dist
+      previous = point
+
+  # Why ?
```

Comment:
For minimally invasive surgical interventions, you want to know the optimal viewing angle that you need to set your C-arm to. You can just write in the comment what you think ("I could not see the usefulness of the slice orientation in current surgery practice. I added this information because it was easily available."). The next person who will come and improve this part will then understand what the motivation was and how the function can be modified.

Reply:
Function has been commented.